File: L011/anyBaseToBase.cpp
#include<iostream>
using namespace std;
  
int abtd(int n, int a) {

        int sum = 0;
        int p = 1;
        while (n != 0) {
            int r = n % 10;
            n = n / 10;

            sum += r * p;
            p *= a;
        }

        return sum;
    }
    
    
    int dtab(int n, int a) {

        int sum = 0;
        int p = 1;
        while (n != 0) {
            int r = n % a;
            n = n / a;

            sum += r * p;
            p *= 10;
        }

        return sum;
    }
  int  main() {
      
      int n;
      cin>>n;
      int sourceBase;
      cin>>sourceBase;
     int  destBase;
     cin>>destBase;
     
     int ans = abtd(n,sourceBase);
     cout<<dtab(ans,destBase);

     return 0;
     
   }   

File: L011/anyBaseToBase.java
import java.util.*;
  
  public class anyBaseToAnyBase{
      public Scanner sc = new Scanner(System.in);

     public static int abtd(int n, int a) {

        int sum = 0;
        int p = 1;
        while (n != 0) {
            int r = n % 10;
            n = n / 10;

            sum += r * p;
            p *= a;
        }

        return sum;
    }
    
    
     public static int dtab(int n, int a) {

        int sum = 0;
        int p = 1;
        while (n != 0) {
            int r = n % a;
            n = n / a;

            sum += r * p;
            p *= 10;
        }

        return sum;
    }
  public static void main(String[] args) {
      Scanner scn = new Scanner(System.in);
      int n = scn.nextInt();
      int sourceBase = scn.nextInt();
     int  destBase= scn.nextInt();
     
     int ans = abtd(n,sourceBase);
     System.out.println(dtab(ans,destBase));
     
   }   
  }